mmcv/ops/nms.py

```diff
@@ -191,17 +226,10 @@ def soft_nms(boxes,
         dets, inds, num_out = ext_module.softnms(*indata_list, **indata_dict)
         inds = inds[:num_out]
     else:
-        dets = boxes.new_empty((boxes.size(0), 5), device='cpu')
-        inds = ext_module.softnms(
-            boxes.cpu(),
-            scores.cpu(),
-            dets.cpu(),
-            iou_threshold=float(iou_threshold),
-            sigma=float(sigma),
-            min_score=float(min_score),
-            method=method_dict[method],
-            offset=int(offset))
-    dets = dets[:inds.size(0)]
```

Comment:
Here should be compatible with `parrots`, another backend  Sensetime. It seems wrong. `dets` from new pytorch soft_nms are sliced but, from parrots are not. We should add sliced dets `dets = dets[:inds.size(0)]`at the end of line 227.